lib/rules/no-inner-compare.js

```diff
@@ -15,27 +15,27 @@ var HINTS = {
 
 module.exports = function(context) {
   return {
-    ExpressionStatement: function(node) {
-      var expression = node.expression;
+    ExpressionStatement(node) {
+      const {expression} = node;
```

Comment:
I would prefer to use `let` instead of `const` unless we're talking about actual singleton constants (those at the top-level module scope that already use uppercase names) or imports

Reply:
I can do that.

But in case you haven't come across this reasoning, I find that `const`, as the ESLint `prefer-const` rule states, can help with "reducing cognitive load and improving maintainability.", since it makes clearer at a glance what items are counters and such (with `let`) and which are simply new scoped variables.

Reply:
yeah, I know about that reasoning, and I don't buy it... 😅 

the author of the let/const spec explicitly stated that this was not the reason, and yet people use it that way 🤷‍♂

Reply:
I think specs determine what we ought to or have to follow, but we don't have to agree with why we have to follow them. But ok, sure...

Reply:
Also by imports using `const` do you mean `require` statements?